File: angular/bootstrap/test/src/app/accounts-template/account.ts
export class Account {
  id: string;
  name: string;
  email: string;
  telephone: string;
  fax: string;
  website: string;
  vatNumber: string;
  description: string;
  addressId: string;
  address: Address;
  parentAccount: Account;
  billingAccount: Account;
  relationType: RelationType;
  // primaryContact
  // contacts
  // workOrders
}

export class Address {
  id: string;
  street: string;
  number: string;
  city: string;
  postalCode: string;
}

export class RelationType {
  id: string;
  name: string;
}

File: _Output/Projects/WebAPI/CodeGenOutput.API/Models/Account.cs
using FluentValidation;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace CodeGenOutput.API.Models
{
    public class Account : Auditable
    {
        public Account()
        {
            this.Contacts = new List<AccountContact>();
        }

        [Key]
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Telephone { get; set; }
        public string Email { get; set; }
        public string Website { get; set; }
        public string VAT { get; set; }

        public Guid? AddressId { get; set; }
        public Address Address { get; set; }

        public ICollection<AccountContact> Contacts { get; set; }
    }

    public class AccountValidator : AbstractValidator<Account>
    {
        public AccountValidator()
        {
            RuleFor(x => x.Name)
                .NotEmpty().WithMessage("Name is required")
                .MaximumLength(100).WithMessage(string.Format("Name has a {0} character limit", 100));
        }
    }
}
